src/Common/tests/System/Net/Http/LoopbackServer.cs

```diff
@@ -147,6 +147,29 @@ public static Task CreateClientAndServerAsync(Func<Uri, Task> clientFunc, Func<L
             return lines;
         }
 
+        public static string GetRequestHeaderValue(List<string> headers, string name)
+        {
+            foreach (string line in headers)
+            {
+                string[] tokens = line.Split(':');
+                if (name.Equals(tokens[0], StringComparison.InvariantCultureIgnoreCase))
+                {
+                    return tokens[1].Trim();
```

Comment:
Is this going to work as expected when headers contain ':' in the value? ie, if we are using digest authentication and include a uri:
```
Authorization: Digest username="user",realm="http-auth@example.org",uri="http://test.com"
```
With the code below this on line 157, I would expect everything after the second ':' to be cut off.